### PR TITLE
ENH: Add option to change data dtype

### PR DESCRIPTION
**Pull request by mgxd**
Adds `data_dtype` option to `DerivativesDataSink` to allow coercing data into desired dtypes.

Inspired by poldracklab/fmriprep#2001

### CODE REVIEW

File: niworkflows/interfaces/bids.py [hunk 1]
```diff
@@ -464,34 +466,45 @@ def _run_interface(self, runtime):
             self._results['out_file'].append(out_file)
             self._results['compression'].append(_copy_any(fname, out_file))
 
-            is_nii = out_file.endswith('.nii') or out_file.endswith('.nii.gz')
-            if self.inputs.check_hdr and is_nii:
+            is_nii = out_file.endswith(('.nii', '.nii.gz'))
+            if is_nii and any((self.inputs.check_hdr, self.inputs.data_dtype)):
```
Comment (oesteban) [inline niworkflows/interfaces/bids.py:470]:
```suggestion
            is_nii = out_file.endswith(('.nii', '.nii.gz'))
            data_dtype = DEFAULT_DTYPES[self.inputs.suffix]
            if isdefined(self.inputs.data_dtype):
                data_dtype = self.inputs.data_dtype
            if is_nii and any((self.inputs.check_hdr, self.inputs.data_dtype)):
```

where
```Python
from collections import defaultdict

def _none():
     return None

DEFAULT_DTYPES = defaultdict({
    "mask": "uint8",
    "dseg": "int16",
    "probseg": "float32",
}, _none)
```

File: niworkflows/interfaces/bids.py [hunk 2]
```diff
@@ -464,34 +466,45 @@ def _run_interface(self, runtime):
...
+                    if curr_codes != xcodes or curr_units != units:
+                        self._results['fixed_hdr'][i] = True
+                        hdr.set_qform(nii.affine, xcodes[0])
+                        hdr.set_sform(nii.affine, xcodes[1])
+                        hdr.set_xyzt_units(*units)
+
+                        # Rewrite file with new header
+                        overwrite_header(nii, out_file)
+
+                if self.inputs.data_dtype:
```
Comment (oesteban) [inline niworkflows/interfaces/bids.py:500]:
```suggestion
                if data_dtype is not None:
```

File: niworkflows/interfaces/bids.py [hunk 3]
```diff
@@ -464,34 +466,45 @@ def _run_interface(self, runtime):
...
+                    if self.inputs.check_hdr:
+                        # load updated NIfTI
+                        nii = nb.load(out_file, mmap=False)
+                    data_dtype = np.dtype(self.inputs.data_dtype)
```
Comment (oesteban) [inline niworkflows/interfaces/bids.py:504]:
```suggestion
                    data_dtype = np.dtype(data_dtype)
```

File: niworkflows/interfaces/bids.py (hunk 1 above)
Comment (mgxd) [reply niworkflows/interfaces/bids.py]: i've incorporated this, but would like to still include the input field `data_dtype` to allow overriding this JIC

Comment (mgxd) [reply niworkflows/interfaces/bids.py]: @oesteban did the latest commit address your concerns?

**Pull request merged**